integration-test/experimental-react/server.js

```diff
@@ -0,0 +1,101 @@
+import express from "express";
+import { renderToPipeableStream } from "react-dom/server";
+import { readFile } from "node:fs/promises";
+
+// Constants
+const isProduction = process.env.NODE_ENV === "production";
+const port = process.env.PORT || 5173;
+const base = process.env.BASE || "/";
+const ABORT_DELAY = 10000;
+
+// Create http server
+const app = express();
+
+// Add Vite or respective production middlewares
+let vite;
+let bootstrapModules = [];
+let assets = [];
+console.log({ isProduction });
+if (!isProduction) {
+  const { createServer } = await import("vite");
+  vite = await createServer({
+    server: { middlewareMode: true, hmr: true },
+    appType: "custom",
+    base,
+  });
+  app.use(vite.middlewares);
+} else {
+  const compression = (await import("compression")).default;
+  const sirv = (await import("sirv")).default;
+  app.use(compression());
+  app.use(base, sirv("./dist/client", { extensions: [] }));
+  const index = await readFile("./dist/client/index.html", "utf-8");
+  for (const script of index.matchAll(
+    /<script type="module" \w+ src="(.*)">/g
+  )) {
+    bootstrapModules.push(script[1]);
+  }
+  for (const link of index.matchAll(
+    /<link rel="stylesheet" \w+ href="(.*)">/g
+  )) {
+    assets.push(link[1]);
+  }
+}
```

Comment:
I know this is meant to be more of a "demo" (ish), but any chance we could flip the conditional here? I always find it easier with `if/else` to use the positive connotation first.

Rather than:
```ts
if (!isProduction) {
  // other envs
} else {
  // production builds
}
```

Use this:
```ts
if (isProduction) {
  // production stuff
} else {
  // other environments
}
```

This keeps the "production" configuration closer to the conditional test for it, rather than breaking it apart.

Reply:
This is based on the [Vite React SSR template](https://github.com/bluwy/create-vite-extra/blob/master/template-ssr-react/server.js#L22) and I'm trying to keep the diff as small as possible, so while I agree and would really like to swap it, it's probably better we don't :)

Reply:
I didn't realize these were copy-pasted so makes sense! Thanks for pointing me to that.